tomviz/DataSource.cxx

```diff
@@ -866,14 +858,10 @@ QVector<double> DataSource::getTiltAngles() const
 
 void DataSource::setTiltAngles(const QVector<double>& angles)
 {
-  vtkDataArray* tiltAngles = this->Internals->TiltAngles;
-  vtkAlgorithm* tp = algorithm();
-  vtkDataObject* data = tp->GetOutputDataObject(0);
-  vtkFieldData* fd = data->GetFieldData();
-  if (fd->GetArray("tilt_angles") != this->Internals->TiltAngles) {
-    tiltAngles = fd->GetArray("tilt_angles");
-  }
-  if (tiltAngles) {
+  auto data = this->dataObject();
+  auto fd = data->GetFieldData();
+  if (fd->GetArray("tilt_angles")) {
```

Comment:
Should this use HasArray?  Or store the result and check if tiltAngles is non-null to avoid calling it twice?

Reply:
I agree, ```HasArray(...)``` is probably better here.